### PR TITLE
Add XFeat to GlueFactory

### PR DESCRIPTION
**Pull request by hardik01shah**
Add [XFeat](https://arxiv.org/abs/2404.19174) as a GlueFactory model that enables XFeat inference, supporting evaluation on point metrics and downstream tasks like Homography Estimation and Pose Estimation on Hpatches and MegaDepth respectively.

### CODE REVIEW

**Review by FelixMoeller3 — APPROVED** [review]
Looks good! We can merge imo

File: gluefactory/models/extractors/xfeat.py [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+        if self.conf.preprocess:
+            image, rh, rw = self.preprocess_tensor(image)
+
+        pred = self.net(image)
+
+        if self.conf.get_sparse_outputs:
+            pred_sparse = self.get_sparse_outputs(data, pred, rh, rw)
+            pred = {**pred, **pred_sparse}
+        
+        return pred
```
Comment (FelixMoeller3) [inline gluefactory/models/extractors/xfeat.py:395]: Could you make sure we return the same keys (names and content) as with other point methods? i.e. keypoints, keypoints_scores and descriptors

**Review by FelixMoeller3 — CHANGES_REQUESTED** [review]

File: gluefactory/models/extractors/xfeat.py [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+    default_conf = {
+        "model_name": ModelName.XFEAT.value,
+        "max_num_keypoints": 4096,
+        "detection_threshold": 0.05,
+        "pretrained": True,
+        "get_sparse_outputs": True,
+        "force_num_keypoints": True,
+        "NMS": {
+            "threshold": 0.05,
+            "kernel_size": 5
```
Comment (RaphaelKreft) [inline gluefactory/models/extractors/xfeat.py:212]: I think we are currently using nms of 3, should we do equal nms for all methods or keep them individual?
Also @hardik01shah make nms option explicit in demo config please and add a comment we are using the sparse xfeat.

**Review by RaphaelKreft — APPROVED** [review]
Overall lgtm. Please address the one comment, but then you can merge

File: gluefactory/models/extractors/xfeat.py (hunk 1 above)
Comment (hardik01shah) [reply gluefactory/models/extractors/xfeat.py:395]: You can check it at Line 363

File: gluefactory/models/extractors/xfeat.py (hunk 2 above)
Comment (hardik01shah) [reply gluefactory/models/extractors/xfeat.py:212]: No, I think NMS should be method dependent.

File: gluefactory/models/extractors/xfeat.py (hunk 1 above)
Comment (FelixMoeller3) [reply gluefactory/models/extractors/xfeat.py:395]: Yeah, I saw it now :D sorry, my bad

**Review by FelixMoeller3 — APPROVED** [review]

**Pull request merged**